DE translation for custom_completions.md (#1056)
nushell/nushell.github.io: de/book/custom_completions.md

die Vervollständigungen für Positions- sowie Markierungs-Argumente enthalten.
Die eigenen Vervollständigungen funktionieren für eigene Befehle und [known external, or `extern`, commands](externs.md).

Eigene Vervollständigungen bestehen aus zwei Teilen:
Dem Befehl der sich um die Vervollständigung kümmert und die Verbindung dieses Befehls mit einem anderne Befehl mit Hilfe von `@`.

## Beispiel eigene Vervollständigung

Hier ein Beispiel:

```
> def tiere [] { ["katze", "hund", "aal" ] }
> def my-command [tier: string@tiere] { print $tier }
>| my-command
katze                 hund                 aal
```

In der ersten Zeile wird ein eigener Befehl erstellt, der eine Liste von drei verschiedenen Tieren zurückgibt.
Dies sind die Werte, die in der Vervollständigung verwendent werden. Ist dieser Befehl erstellt, kann er für weitere Vervollständigungen von Befehlen und [`extern`](/commands/docs/extern.md) verwendet werden.

In der zweiten Zeile, wird `string@tiere` verwendet.
Dies sagt Nushell zwei Dinge:
Die Form des Arguments um den Typ überprüfen zu können, sowie die Vervollständigung, falls der Benutzer diese an der Stelle verwenden möchte.

> Auf der dritten Zeile wird der Name der Vervollständigung `my-command` eingegeben gefolgt von der `<tab>` Taste. Dies führt die Vervollständigung aus. Eigene Vervollständigungen funktionieren identisch zu anderen Vervollständigungen. Wird `a` gefolgt von der `<tab>` Taste gedrückt, wird automatisch "aal" ausgegeben.

Reviewer: ```suggestion
Auf der dritten Zeile wird der Name des zu vervollständigenden Befehls `my-command` eingegeben, gefolgt von der `<tab>` Taste. Dies führt die Vervollständigung aus. Eigene Vervollständigungen funktionieren identisch zu anderen Vervollständigungen. Wird `a` gefolgt von der `<tab>` Taste gedrückt, wird automatisch "aal" ausgegeben.
```

    ] {
        print $"Als ($tier) heisse ich ($name)."
    }
}
```
Der Befehl `tier-name` gibt die entsprechende Liste der Namen zurück. Dies funktioniert, weil der Wert der `$context` Variablen, dem Text entspricht, der bis zu dem Zeitpunkt eingegeben wurde.

```
>| my-command
katze                 hund                 aal
>| my-command hund
Lulu                Enzo
>my-command hund enzo
Als hund heisse ich Enzo
```

Auf der zweiten Zeile wird, sobald die `<tab>` Taste gedrückt wurde, das Argument `"my-command hund"` dem `tier-namen` Kontext übergeben.

## Eigene Vervollständigungen und [`extern`](/commands/docs/extern.md)

Sehr mächtig ist die Kombination von eigenen Vervollständigungen mit [externen Befehlen](externs.md). Diese funktionieren gleich wie die Vervollständigungen zu eigenen Befehlen:
Erstellen der Vervollständigung und anbinden an ein Positions- oder Markierungs-Argument des `extern` mit dem `@` Zeichen.

In der Default Konfiguration finden sich ebenfalls Vervollständigungen:

```
export extern "git push" [
    remote?: string@"nu-complete git remotes",  # the name of the remote
    refspec?: string@"nu-complete git branches" # the branch / refspec
    ...
]
```

> Die Vervollständigung erfüllt hier die gleich Rolle wie in den Beispielen zuvor. Es werden zwei verschiedene Vervollständigungen verwendet, abhängig von der Position, die bisher eingegeben wurde.

Reviewer: ```suggestion
Die Vervollständigung erfüllt hier die gleiche Rolle wie in den Beispielen zuvor. Es werden zwei verschiedene Vervollständigungen verwendet, abhängig von der Position, die bisher eingegeben wurde.
```

## Eigene Beschreibungen

Alternativ zu einer Liste von Strings, kann eine Vervollständigung auch einen Record aus einem `value` und einer `description` zurückgeben.

```
def my_commits [] {
    [
        { value: "5c2464", description: "Add .gitignore" },
        { value: "f3a377", description: "Initial commit" }
    ]
}
```

> **Notiz**
>
> Mit dem folgenden Schnippsel:
>
> ```nu
> def my-command [commit: string@my_commits] {
>     print $commit
> }
> ```
> 
> aufgepasst, auch wenn die Vervollständigung folgendes zeigt
>
> ```nu
> >_ my-command <TAB>
> 5c2464  Add .gitignore
> f3a377  Initial commit
> ```
>
> nur die Werte, hier "5c2464" und "f3a377", werden in Befehls Argumenten verwendet!

## Extere Vervollständigungen

Externe Vervollständigungen können ebenfalls integriert werden, anstatt derer von Nushell.

> Dafür muss das Feld `external_completer` in `config.nu` auf einen [closure](/book/types_of_data.md#closures)) gesetzt werden, welcher ausgewertet wird, wenn keine Nushell Vervollständigungen gefunden werden.

Reviewer: ```suggestion
Dafür muss dem Feld `external_completer` in `config.nu` eine [Closure](/book/types_of_data.md#closures)) übergeben werden, welche ausgewertet wird, wenn keine Nushell Vervollständigungen gefunden werden.
```

```nu
> $env.config.completions.external = {
>     enable: true
>     max_results: 100
>     completer: $completer
> }
```

> Der Closure kann konfiguriert werden, einen externen Vervollständiger wie [carapace](https://github.com/rsteube/carapace-bin) zu verwenden.
> 
> Wenn der Closure einen nicht lesbaren json (z.B. einen leeren String) zurückgibt, fällt Nushell auf die Datei Vervollständigung zurück.

Reviewer: ```suggestion
Die Closure kann konfiguriert werden, einen externen Vervollständiger wie [carapace](https://github.com/rsteube/carapace-bin) zu verwenden.

Wenn die Closure einen nicht lesbaren Wert (z.B. einen leeren String) zurückgibt, fällt Nushell auf die Datei Vervollständigung zurück.
```

Ein externer Vervollständiger ist eine Funktion, die den aktuellen Befehl als String Liste entgegennimmt, und eine Liste von Records mit `value` und `description` zurückgibt. Wie bei eigenen Nushell Vervollständigungen.

> > **Notiz**
> > Dieser Closure nimmt den aktuellen Befehl als Liste entgegen. Zum Beispiel, wird `my-command --arg1 <tab>` als Vervollständigung `[my-command --arg1 " "]` erhalten.

Reviewer: ```suggestion
> **Notiz**
> Diese Closure nimmt den aktuellen Befehl als Liste entgegen. Zum Beispiel, wird `my-command --arg1 <tab>` als Vervollständigung `[my-command --arg1 " "]` erhalten.
```
